firstvoices/backend/views/immersion_label_views.py

```diff
@@ -132,8 +134,13 @@ class ImmersionLabelViewSet(
 
     def get_queryset(self):
         site = self.get_validated_site()
+        visible_dictionary_entry = DictionaryEntry.objects.visible(
+            self.request.user
+        ).filter(immersion_labels=OuterRef("pk"))
```

Comment:
Another, somewhat simpler way to accomplish this is to use a condition when prefetching the dictionary entry. We do something similar for the site features in the Sites API (those are filtered by value not visibility but it shows the method).